src/ui/public/agg_types/buckets/date_histogram.js

```diff
@@ -122,9 +117,10 @@ export const dateHistogramBucketAgg = new BucketAggType({
       },
       write: function (agg, output) {
         setBounds(agg);
```

Comment:
this can be removed as getTimeBuckets already calls setBounds ?

and if getTimeBuckets is the one calling setBounds, we can pass the timeBucket to the setBounds function amd we do't need to call getTimeBuckets from it again ... hopefuly solves this circular weirdness ?